public/tutorials/using-infura-custom-provider.md

```diff
@@ -1,136 +1,152 @@
 
-engine.addProvider(new WalletSubprovider(wallet, {}));
-engine.addProvider(new Web3Subprovider(new Web3.providers.HttpProvider(providerUrl)));
-engine.start(); // Required by the provider engine.
-```
+3. Add a Ropsten network definition:
 
-Finally, we want to export our Truffle configuration. Here we use this wallet only when we want to deploy to the ropsten network:
-
-```javascript
-module.exports = {
-  networks: {
-    "ropsten": {
-      network_id: 3,    // Official ropsten network id
-      provider: engine, // Use our custom provider
-      from: address     // Use the address we derived
-    }
-  },
-  rpc: {
-    // Use the default host and port when not using ropsten
-    host: "localhost",
-    port: 8545
-  }
-};
-```
+   ```javascript
+   module.exports = {
+     networks: {
+       ropsten: {
+         provider: new HDWalletProvider(mnemonic, "https://ropsten.infura.io/<INFURA_Access_Token>"),
+         network_id: 3
+       }   
+     }
+   };
+   ```
 
-### Discussion
-
-In general, this is a lot of work in order to use an hd wallet with Truffle. This is a very early feature, and we'd highly encourage anyone to create a new library that encapsulates the above code.
-
-### Full Code
-
-```javascript
-var bip39 = require("bip39");
-var hdkey = require('ethereumjs-wallet/hdkey');
-var ProviderEngine = require("web3-provider-engine");
-var WalletSubprovider = require('web3-provider-engine/subproviders/wallet.js');
-var Web3Subprovider = require("web3-provider-engine/subproviders/web3.js");
-var Web3 = require("web3");
-const FilterSubprovider = require('web3-provider-engine/subproviders/filters.js');
-
-// Get our mnemonic and create an hdwallet
-var mnemonic = "couch solve unique spirit wine fine occur rhythm foot feature glory away";
-var hdwallet = hdkey.fromMasterSeed(bip39.mnemonicToSeed(mnemonic));
-
-// Get the first account using the standard hd path.
-var wallet_hdpath = "m/44'/60'/0'/0/";
-var wallet = hdwallet.derivePath(wallet_hdpath + "0").getWallet();
-var address = "0x" + wallet.getAddress().toString("hex");
-
-var providerUrl = "https://testnet.infura.io";
-var engine = new ProviderEngine();
-// filters
-engine.addProvider(new FilterSubprovider());
-
-engine.addProvider(new WalletSubprovider(wallet, {}));
-engine.addProvider(new Web3Subprovider(new Web3.providers.HttpProvider(providerUrl)));
-engine.start(); // Required by the provider engine.
-
-module.exports = {
-  networks: {
-    "ropsten": {
-      network_id: 3,    // Official ropsten network id
-      provider: engine, // Use our custom provider
-      from: address     // Use the address we derived
-    }
-  },
-  rpc: {
-    // Use the default host and port when not using ropsten
-    host: "localhost",
-    port: 8545
-  }
-};
-```
+   Things to notice:
+
+   * While the example has only a single network defined, you can define multiple networks as normal.
+
+   * The `provider` for the `ropsten` network definition instantiates the `HDWalletProvider`.
```

Comment:
Should probably add a line here about why `provider` is a function (see comment above)